packages/web3/test/integration/main_web3.test.ts

```diff
@@ -0,0 +1,303 @@
+		'Create Web3 class instance with ws string providers',
+		() => {
+			it('should create instance with string of ws provider', async () => {
+				web3 = new Web3(clientUrl);
+				expect(web3).toBeInstanceOf(Web3);
+			});
+
+			itIf(
+				process.env.INFURA_GOERLI_WS
+					? process.env.INFURA_GOERLI_WS.toString().includes('ws')
+					: false,
+			)('should create instance with string of external ws provider', async () => {
+				web3 = new Web3(process.env.INFURA_GOERLI_WS!);
+				// eslint-disable-next-line jest/no-standalone-expect
+				expect(web3).toBeInstanceOf(Web3);
+			});
+		},
+	);
+	describe('Web3 providers', () => {
+		it('should set the provider', async () => {
+			web3 = new Web3('http://dummy.com');
+
+			web3.provider = clientUrl;
+
+			expect(web3).toBeInstanceOf(Web3);
+
+			const response = await web3.eth.getBalance(accounts[0]);
+
+			expect(response).toMatch(/0[xX][0-9a-fA-F]+/);
+		});
+
+		it('setProvider', async () => {
+			let newProvider: Web3BaseProvider;
+			web3 = new Web3('http://dummy.com');
+			if (clientUrl.startsWith('http')) {
+				newProvider = new Web3.providers.HttpProvider(clientUrl);
+			} else {
+				newProvider = new Web3.providers.WebsocketProvider(clientUrl);
+			}
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-call
+			web3.setProvider(newProvider as SupportedProviders<Web3EthExecutionAPI>);
+
+			expect(web3.provider).toBe(newProvider);
+		});
+
+		it('providers', async () => {
+			const res = Web3.providers;
+
+			expect(Web3.providers.HttpProvider).toBe(HttpProvider);
+			expect(res.WebsocketProvider).toBe(WebsocketProvider);
+			expect(res.IpcProvider).toBe(IpcProvider);
+		});
+
+		it('currentProvider', async () => {
+			web3 = new Web3(clientUrl);
+
+			let checkWithClass;
+			if (clientUrl.startsWith('ws')) {
+				checkWithClass = Web3.providers.WebsocketProvider;
+			} else if (clientUrl.startsWith('http')) {
+				checkWithClass = Web3.providers.HttpProvider;
+			} else {
+				checkWithClass = Web3.providers.IpcProvider;
+			}
+			expect(web3.currentProvider).toBeInstanceOf(checkWithClass);
+		});
+
+		it('givenProvider', async () => {
+			const { givenProvider } = web3;
+			expect(givenProvider).toBeUndefined();
+		});
+	});
+
+	describe('Module instantiations', () => {
+		it('should create contract', () => {
+			const basicContract = new web3.eth.Contract(BasicAbi);
+			expect(basicContract).toBeInstanceOf(Contract);
+		});
+	});
+
+	describe('Batch Request', () => {
+		let request1: JsonRpcOptionalRequest;
+		let request2: JsonRpcOptionalRequest;
+		beforeEach(() => {
+			request1 = {
+				id: 10,
+				method: 'eth_getBalance',
+				params: [accounts[0], 'latest'],
+			};
+			request2 = {
+				id: 11,
+				method: 'eth_getBalance',
+				params: [accounts[1], 'latest'],
+			};
+		});
+
+		it('should execute batch requests', async () => {
+			web3 = new Web3(clientUrl);
+
+			// eslint-disable-next-line @typescript-eslint/no-floating-promises, @typescript-eslint/no-unsafe-call
```

Comment:
Please don't disable `floating-promises` lint warnings. These are really evil in JS code and can have unseen side effects.